LS-446-remove-replicated (#33)
datastax/pulsar-docs: modules/ROOT/pages/quickstart-helm-installs.adoc

:page-tag: luna-streaming,dev,admin,install,pulsar

> You have three options for installing DataStax Luna Streaming (Pulsar):

Reviewer: ```suggestion
You have options for installing DataStax Luna Streaming:
```

You have two options for installing DataStax Luna Streaming (Pulsar):

* Via the provided Helm chart for an existing Kubernetes environment locally or with a cloud provider, as covered in this topic. 
* Via Pulsar tarball for deployment to a single server/VM, or to multiple servers/VMs. See xref:quickstart-server-installs.adoc[Quick Start for Server/VM installs].

The Helm chart and options described below configure an Apache Pulsar cluster.
It is designed for production use, but can also be used in local development environments with the proper settings.

> https://helm.sh[Helm] version 3 must be installed and initialized to use the chart. Version 2 of Helm is not supported.

Reviewer: ```suggestion
The resulting configuration includes support for:

* xref:quickstart-helm-installs.adoc#_tls[TLS]
* xref:quickstart-helm-installs.adoc#_authentication[Authentication]
* WebSocket Proxy
* Standalone Functions Workers
* Pulsar IO Connectors
* xref:quickstart-helm-installs.adoc#_tiered_storage[Tiered Storage] including Tardigarde distributed cloud storage
* xref:quickstart-helm-installs.adoc#_pulsar_sql[Pulsar SQL Workers]
* Pulsar Admin Console for managing the cluster
* Pulsar heartbeat
* Burnell for API-based token generation
* Prometheus, Grafana, and Alertmanager stack with default Grafana dashboards and Pulsar-specific alerting rules
* cert-manager with support for self-signed certificates as well as public certificates using ACME; such as Let's Encrypt
* Ingress for all HTTP ports (Pulsar Admin Console, Prometheus, Grafana, others)

https://helm.sh[Helm] v3 must be installed and initialized to use the chart. Helm v2 is not supported.
```

== Prerequisites 

> * Helm version 3

Reviewer: ```suggestion
* https://helm.sh/docs/[Helm v3]
```

* A Kubernetes cluster
* No less than 16 gigs of RAM and 8 virtual processor cores (4 physical cores)

> For details on Helm, refer to its https://helm.sh/docs/[documentation].
> For details on Minikube, see its https://minikube.sigs.k8s.io/docs/start/[documentation]. 

Reviewer: ```suggestion
```

> If you have specific storage class requirement, for example fixed IOPS disks in AWS, you can have the chart configure the storage classes for you.
> Here are examples from the cloud providers:

Reviewer: ```suggestion
If you have specific storage class requirement, for example fixed IOPS disks in AWS, you can have the chart configure the storage classes for you.

Here are examples from the cloud providers:
```

> #  fsType: ext4

Reviewer: Remove `# ` from lines 47 and 48.

> #  fsType: ext4
> #  extraParams:
> #     iopsPerGB: "10"

Reviewer: ```suggestion
 fsType: ext4
 extraParams:
    iopsPerGB: "10"
```

----

==== Google Cloud Provider
----

> # For GCP
> # default_storage:
> #   provisioner: kubernetes.io/gce-pd
> #   type: pd-ssd
> #   fsType: ext4
> #   extraParams:
> #      replication-type: none

Reviewer: ```suggestion
default_storage:
  provisioner: kubernetes.io/gce-pd
  type: pd-ssd
  fsType: ext4
  extraParams:
     replication-type: none
```

----

==== Azure 

----

> # For Azure

Reviewer: Remove `# ` in Azure file.

`kubectl config set-context $(kubectl config current-context) --namespace=pulsar`

> Once Pulsar is installed, continue to access your cluster. 

Reviewer: ```suggestion
Once Pulsar is installed, you can now access your Luna Streaming cluster. 
```

`kubectl get service -n pulsar`

> 

Reviewer: ```suggestion

== Accessing the Pulsar cluster on localhost

To port forward the proxy admin and Pulsar ports to your local machine:

`kubectl port-forward -n pulsar $(kubectl get pods -n pulsar -l component=proxy -o jsonpath='{.items[0].metadata.name}') 8080:8080`

`kubectl port-forward -n pulsar $(kubectl get pods -n pulsar -l component=proxy -o jsonpath='{.items[0].metadata.name}') 6650:6650`

Or if you would rather go directly to the broker:

`kubectl port-forward -n pulsar $(kubectl get pods -n pulsar -l component=broker -o jsonpath='{.items[0].metadata.name}') 8080:8080`

`kubectl port-forward -n pulsar $(kubectl get pods -n pulsar -l component=broker -o jsonpath='{.items[0].metadata.name}') 6650:6650`

```

`enableTokenAuth: yes`

[#tls]

> == TLS
> 

Reviewer: ```suggestion
```

> === Automatically generating certificates using cert-manager
> 

Reviewer: ```suggestion
```

> === Manually configuring certificate secrets for TLS

Reviewer: ```suggestion
== Manually configuring certificate secrets for TLS
```

`enableTls: yes`

> == Next

Reviewer: ```suggestion
== What's next?
```
